app/services/queries/reviews_turnaround_per_user_project_query.rb

```diff
@@ -0,0 +1,32 @@
+module Queries
+  class ReviewsTurnaroundPerUserProjectQuery < BaseService
+    attr_reader :time_interval
+
+    def initialize(time_interval:)
+      @time_interval = time_interval
+    end
+
+    def call
+      run
+    end
+
+    private
+
+    def query
+      Events::Review.joins(:review_request, pull_request: :project).includes(:review_request)
+                    .where(opened_at: @time_interval.starting_at..@time_interval.ending_at)
+    end
+
+    def run
+      query.collect do |review|
```

Comment:
`collect` will be fetching all the records in memory at once, never tried it but maybe the `lazy` here https://collectiveidea.com/blog/archives/2015/03/05/optimizing-rails-for-memory-usage-part-3-pluck-and-database-laziness is what you need?

Reply:
https://ruby-doc.org/core-2.5.0/Enumerator/Lazy.html

Reply:
Hey! Thank you for making me realize of that! I am going to try that approach.